bindata/v4.1.0/alerts/cpu-utilization.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: cpu-utilization
+  namespace: openshift-kube-apiserver
+  annotations:
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    exclude.release.openshift.io/internal-openshift-hosted: "true"
+spec:
+  groups:
+    - name: control-plane-cpu-utilization
+      rules:
+        - alert: HighOverallControlPlaneCPU
```

Comment:
I understand where you are coming from but a couple of things:

With CPU being a compressible resource it's very hard to write an alert that will actually be correct, as we see in CI, this alert fires but clearly everything went well as CI passed otherwise no? 

If there is CPU pressure it will just slow things down not stop working entirely? If yes inn which case a latency alert should fire for each component on the control plane.

What I would suggest instead is to write alerts for each control plane component and alert on that so as suggested latency alerts, error alerts, etc. For example, you could write alerts like:
* scheduling latency
* deployment controller reconcile loops taking long
* apiserver latency

This is very much [cause vs symptoms based alerting.](https://prometheus.io/docs/practices/alerting/#what-to-alert-on) The control plane alerts (symptoms) suggested above will also be able to catch other causes, not just high CPU utilization.

This might be a good read https://www.robustperception.io/when-to-alert-with-prometheus

Reply:
We see cases of clusters that are running along just fine.  All latencies are reasonable.  All external signs show the cluster is perfectly healthy.  But they are time bombs.  They're using say 80% of the CPU on their system.  When one shuts down, we get a cascading failure.

We want to provide a warning before the issue becomes catastrophic, but no "this loop is taking too long" check will ever come back positive.  Such a warning can tell you that you're in the middle of a crisis, but provides no value in terms of predicting that you're going to have a crisis.